src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs

```diff
@@ -549,8 +549,7 @@ static void RemoveUnsupportsOnDifferentPlatforms(SmallDictionary<string, Version
 
             static void RemoveUnsupportedWithLessVersion(Version supportedVersion, Versions attribute)
             {
-                if (attribute.UnsupportedFirst != null &&
-                    attribute.UnsupportedFirst <= supportedVersion)
+                if (supportedVersion.IsGreaterThanOrEqualTo(attribute.UnsupportedFirst))
                 {
                     attribute.UnsupportedFirst = null;
                 }
```

Comment:
Behavior when UnsupportedFirst is null:

- Old code, condition not met.
- New code, condition is met.

Should the null check stays?

Reply:
Here the null check was not required (added that because it was causing NRE)